source/develop/release-management/features/virt/nvdimm-device.md

```diff
@@ -0,0 +1,67 @@
+
+The NVDIMM device in the guest is emulated so its mode, partitioning, etc. can be reconfigured in the guest independently of the host device settings.
+
+Information about the emulation details can be found in
+[QEMU NVDIMM documentation](https://github.com/qemu/qemu/blob/master/docs/nvdimm.txt).
+
+## Prerequisites
+
+NVDIMM hardware is needed to test this feature.  To a limited extent, QEMU emulated devices in VMs used as hosts can be used for testing during implementation.
+
+## Limitations
+
+Memory snapshots are disabled when an NVDIMM device is present in a VM.  There is no way to make a snapshot of NVDIMM content and a memory snapshot cannot work correctly without having the corresponding NVDIMM data.  If a memory snapshot is attempted, a non-memory snapshot is created instead.
+
+In oVirt, each NVDIMM device passed to a VM has an automatically assigned label area of a fixed size of 128 KB.  Having a label is required on POWER and 128 KB is the minimum label size allowed by QEMU.
+
+The guest device size is not user configurable.  If the whole host NVDIMM device size should not be used, the device should be partitioned accordingly on the host and the corresponding partition should be used.
+
+The guest device size may be a bit lower to be in compliance with libvirt and QEMU alignment and size adjustments.  Precise sizing is also needed in order to make memory hot plug work.
+
+libvirt and QEMU perform their own size and label placement adjustments.  If those internal arrangements ever get changed, data loss may be experienced.
+
+NVDIMM hot plug is not supported.
+
+VMs with NVDIMM devices must be pinned to a host and cannot be migrated.
+
+## User Experience
+
+An NVDIMM device is added or removed to or from the VM the same way as a host device.  There is special NVDIMM category of host devices containing NVDIMM devices available on the host.
+
+TODO: Add a screenshot.
+
+## Implementation considerations
+
+Vdsm must report NVDIMM devices present on the host and their attributes (path, mode, size, alignment, ...).  Vdsm uses ndctl tool for that purpose and is dependent on the content of its JSON output.  Contingent changes to the ndctl output could break the functionality in future.
+
+NVDIMM size computation on the Engine side is dependent on libvirt and QEMU internal size adjustments in order to ensure that:
+
+- It can fit into the size of the backing device (which is a bit more complicated than just using size <= size of the backing device).
+- It is properly aligned in the end result (to the architecture minimum DIMM size) to make regular memory hot plug work in the guest OS.
+
+If the libvirt or QEMU internal adjustments ever change, it can break, misplace the label or trim data, possibly resulting in data loss.  There is currently no way to deal with that problem.
```

Comment:
Can you please add also a section on how to test the feature?

Reply:
Done.